class-fp-fetch-remote-posts.php

```diff
@@ -0,0 +1,376 @@
+					if ( isset( $assoc_args['post_type'] ) && ! empty( $assoc_args['post_type'] ) ) :
+						$post_types = $assoc_args['post_type'];
+						/**
+						 * Check if this argument sets for multiple CPTs.
+						 */
+						if ( false !== stripos( $post_types, ',', true ) ) :
+							$post_types = explode( ',', $post_types );
+							if ( ! empty( $post_types ) && is_array( $post_types ) ) :
+								foreach ( $post_types as $post_type ) :
+									$this->frp_import_posts( $url, $post_type );
+								endforeach;
+							endif;
+						else :
+							$this->frp_import_posts( $url, $post_types );
+						endif;
+					else :
+						$this->frp_import_posts( $url, 'posts' );
+					endif;
+				else :
+					/* translators: %s: remote url */
+					WP_CLI::error( sprintf( esc_html__( '%1$s is not a valid URL.', 'fetchremoteposts' ), $url ) );
+				endif;
+			else :
+				WP_CLI::error( 'URL parameter missing.' );
+			endif;
+		}
+
+		/**
+		 * This function imports the posts from the receiving URL and post type.
+		 *
+		 * @param string $url The remote WordPress URL.
+		 * @param string $post_type The post type to be imported.
+		 */
+		protected function frp_import_posts( $url, $post_type ) {
+
+			$remote   = "{$url}/wp-json/wp/v2/{$post_type}?per_page=9";
+			$response = wp_remote_get( $remote );
+
+			if ( is_wp_error( $response ) ) :
+				$error_msg = $response->get_error_message();
+				/* translators: %s: error message */
+				WP_CLI::error( sprintf( esc_html__( 'The API returned an error: %1$s', 'fetchremoteposts' ) ), $error_msg );
+			endif;
+
+			// Get the body.
+			$posts = json_decode( wp_remote_retrieve_body( $response ) );
+
+			// Exit if nothing is returned.
+			if ( empty( $posts ) ) :
+				WP_CLI::error( esc_html__( 'There are no posts in the URL requested.', 'fetchremoteposts' ) );
+			endif;
+
+			// For each post.
+			foreach ( $posts as $post ) :
+
+				esc_html_e( 'Importing posts started..', 'fetchremoteposts' );
+				echo "\n";
+
+				if ( isset( $post->title->rendered ) && ! empty( $post->title->rendered ) ) :
+					$remote_post_id = $post->id;
+					$title          = $post->title->rendered;
+					$content        = ( isset( $post->content->rendered ) && ! empty( $post->content->rendered ) ) ? $post->content->rendered : '';
+					$excerpt        = ( isset( $post->excerpt->rendered ) && ! empty( $post->excerpt->rendered ) ) ? $post->excerpt->rendered : '';
+					$status         = ( isset( $post->status ) && ! empty( $post->status ) ) ? $post->status : '';
+					$post_id        = wp_insert_post(
+						array(
+							'post_title'   => $title,
+							'post_content' => $content,
+							'post_excerpt' => $excerpt,
+							'post_status'  => $status,
+							'post_author'  => 1, // Assigning the post to the site administrator.
+							'post_type'    => $post->type,
+						)
+					);
+
+					// check if the featured image is set.
+					if ( isset( $post->featured_media ) && ! empty( $post->featured_media ) ) :
```

Comment:
You can try `thumbnail_id` post_meta. and it will give you the attachment id of the featured image set for the post.

Add this in the else part here.